sqlglot/expressions.py

```diff
@@ -487,7 +487,7 @@ def flatten(self, unnest=True):
         """
         for node, _, _ in self.dfs(prune=lambda n, p, *_: p and not type(n) is self.__class__):
             if not type(node) is self.__class__:
-                yield node.unnest() if unnest else node
+                yield node.unnest() if unnest and not isinstance(node, Subquery) else node
```

Comment:
This seems like a bug: if one were to call `flatten` on e.g. a `Connector` with a `Subquery` child, the child would also be unnested.

```python
>>> from sqlglot import parse_one
>>> list(parse_one("a and (select true from t where id = 1)").flatten())[1].sql()
'SELECT TRUE FROM t WHERE id = 1'
```